File: backend/semgrep_rules/tests/javascript/kisa_sw_48_dns_security_decision.js
const dns = require("dns");
function unsafe(host, trusted) {
  dns.lookup(host, (err, address) => {
    // ruleid: kisa.sw48.javascript.dns-result-used-for-security-decision-review
    if (address === trusted) console.log("trusted");
  });
}
function safe(verified) {
  // ok: kisa.sw48.javascript.dns-result-used-for-security-decision-review
  return verified;
}

File: backend/semgrep_rules/tests/python/kisa_sw_48_dns_security_decision.py
import socket
def unsafe(host, trusted):
    ip = socket.gethostbyname(host)
    # ruleid: kisa.sw48.python.dns-result-used-for-security-decision-review
    if ip == trusted:
        return True
    return False
def safe(cert_ok):
    # ok: kisa.sw48.python.dns-result-used-for-security-decision-review
    return cert_ok
